### ISSUE TITLE
How do I union queries where the model contains a JSON field?

### ISSUE THREAD

**Issue by dsmurrell** [coleifer/peewee#2437]
If I have two or more queries to join (say q1, q2 and q3) and I'm using set operations on them like: 

`query = q1 & q2` or `query = q2 | q3` or `query = q1 - q3`

I know that there is a limitation that the selected columns for these should be the same, but I'm finding that I get the errors:

`peewee.ProgrammingError: could not identify an equality operator for type json`
`psycopg2.errors.UndefinedFunction: could not identify an equality operator for type json`

whenever all fields are selected and the model includes a `JSONField` from `playhouse.postgres_ext`

Is this any way to resolve this? My use case is that I'm passing a list of queries through to another function where I want to union them and so I don't have the luxury of adding `.where(xyz)` statements onto an exiting query to limit the selection using multiple where filters. Also, the set operations seem more powerful in general as passing a list of where params is quite limiting.

**Comment by dsmurrell** [comment]
Doing a bit of searching, it looks like the UNION operation is trying to take out duplicates and needs to compare the JSON content... is there any way in the union from peewee that I can specify that I'd like only apply distinct on say `id`?

**Comment by coleifer** [comment]
Yes, you need to use UNION ALL if you don't want duplicates removed (which is why you're getting that error). You can use `.union_all()` or the addition operator overload.

**Comment by dsmurrell** [comment]
Thanks, I came across that option in my searches. In the docs, whenever I look, I can't seem to find the docs on query set operations. Am I blind?

I was typing this out below before you responded but then I got consumed by something else:

"Trying out `BinaryJSONField` instead of `JSONField` now and this seems to work, but I'm not sure what the implications are of changing our whole schema to use `BinaryJSONField` instead of `JSONField` everywhere. Are these supposed to be interchangeable or will changing the schema change the data in some way?"
